### PR TITLE
NEW: Search domains internally for checking reservated domains

### PR DESCRIPTION
**Pull request by ilhanmercan**

### CODE REVIEW

File: src/back/domains.py [hunk 1]
```diff
@@ -66,8 +67,28 @@ def is_exist(domain_name='', epp_id=''):
     """
     from back.models.domain import Domain
     if epp_id:
-        return bool(Domain.domains.filter(epp_id=epp_id).first())
-    return bool(Domain.domains.filter(name=domain_name).first())
+        return Domain.domains.filter(epp_id=epp_id).first()
```
Comment (vesellov) [inline src/back/domains.py]: now `is_exist()` will return you domain object instance, instead of boolean... this is not correct.

Comment (vesellov) [reply src/back/domains.py]: we have a `find(domain_name='', epp_id='')` method bellow which is doing the same, why do you want to change `is_exist()` method, @ilhanmercan ?

File: src/back/domains.py [hunk 2]
```diff
@@ -141,6 +162,11 @@ def create(
     return new_domain
 
 
+def delete(domain_id):
+    from back.models.domain import Domain
+    return Domain.domains.filter(id=domain_id).delete()
```
Comment (vesellov) [inline src/back/domains.py]: this will fail if domain with that `domain_id` not exist

Comment (ilhanmercan) [reply src/back/domains.py]: no, it won't fail.
Check this:
```
(Pdb) delete(500)
(0, {'back.Domain': 0})

(Pdb) delete(13)
(1, {'back.Domain': 1})
```

so, if object is not there, it still will go through and won't raise an error.

File: src/back/domains.py (hunk 1 above)
Comment (ilhanmercan) [reply src/back/domains.py]: okay, thanks! I just saw it. going to change it

File: src/back/domains.py (hunk 2 above)
Comment (vesellov) [reply src/back/domains.py]: okay... I see. it is not very obvious. 👍

**Pull request merged**